assets/js/src/subscribers/importExport/import/step_data_manipulation/match_table.jsx

```diff
@@ -1,118 +1,127 @@
 import matchColumns from './match_columns.jsx';
 
 const MAX_SUBSCRIBERS_SHOWN = 10;
 
+function ColumnDataMatch({ header, subscribers }) {
+  const matchedColumnTypes = matchColumns(subscribers, header);
+  // selectedColumns = _.pluck(matchedColumnTypes, 'column_id');
+  return (
+    <tr>
+      <th>{MailPoet.I18n.t('matchData')}</th>
+      {
+        matchedColumnTypes.map((columnType, i) => (
+          // eslint-disable-next-line react/no-array-index-key
+          <th key={columnType.column_id + i}>
+            <select
+              className="mailpoet_subscribers_column_data_match"
+              data-column-id={columnType.column_id}
+              data-validation-rule="false"
+              data-column-index={i}
+              id={`column_${i}`}
+            />
+          </th>
+        ))
+      }
+    </tr>
+  );
+}
+ColumnDataMatch.propTypes = {
+  header: PropTypes.arrayOf(PropTypes.string).isRequired,
+  subscribers: PropTypes.arrayOf(PropTypes.arrayOf(PropTypes.string)).isRequired,
+};
+
+function Header({ header }) {
+  return (
+    <tr className="mailpoet_header">
+      <td />
+      {header.map(headerName => <td key={headerName}>{headerName}</td>)}
+    </tr>
+  );
+}
+Header.propTypes = {
+  header: PropTypes.arrayOf(PropTypes.string).isRequired,
+};
+
+function Subscriber({ subscriber, index }) {
+  return (
+    <>
+      <td>{index}</td>
+      {/* eslint-disable-next-line react/no-array-index-key */}
+      {subscriber.map((field, i) => <td key={field + index + i}>{field}</td>)}
+    </>
+  );
+}
+Subscriber.propTypes = {
+  subscriber: PropTypes.arrayOf(PropTypes.string).isRequired,
+  index: PropTypes.node.isRequired,
+};
+
+function Subscribers({ subscribers, subscribersCount }) {
+  const filler = '. . .';
+  const fillerArray = Array(subscribers[0].length).fill(filler);
+  return (
+    <>
+      {
+        subscribers
+          .slice(0, MAX_SUBSCRIBERS_SHOWN)
+          .map((subscriber, i) => (
+            <tr key={subscriber[0]}>
+              <Subscriber subscriber={subscriber} index={i + 1} />
+            </tr>
+          ))
+      }
+      {
+        subscribersCount > MAX_SUBSCRIBERS_SHOWN + 1
+          ? <tr key="filler"><Subscriber subscriber={fillerArray} index={filler} /></tr>
+          : null
+      }
+      {
+        subscribersCount > MAX_SUBSCRIBERS_SHOWN
+          ? (
+            <tr key={subscribers[subscribersCount - 1][0]}>
+              <Subscriber
+                subscriber={subscribers[subscribersCount - 1]}
+                index={subscribersCount}
+              />
+            </tr>
+          )
+          : null
+      }
+    </>
+  );
+}
+Subscribers.propTypes = {
+  subscribersCount: PropTypes.number.isRequired,
+  subscribers: PropTypes.arrayOf(PropTypes.arrayOf(PropTypes.string)).isRequired,
+};
+
 function MatchTable({
   subscribersCount,
   subscribers,
   header,
 }) {
-  let selectedColumns = [];
+  const selectedColumns = [];
```

Comment:
Maybe I am missing something, but the `selectedColumns` variable seems to not be used anywhere! We assign values to it but never use it.

Reply:
I think so. I was about to do this using react state and props and pass this data the modern way. But in the end, I had to re-use some old jquery code I didn't need to use this variable but forgot to delete it.